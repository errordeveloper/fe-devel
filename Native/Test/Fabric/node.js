F = require('Fabric').createClient();

var mapNamedObjectsToNames = function (namedObjects) {
  var result = {};
  for (var i in namedObjects)
    result[i] = namedObjects[i].getName();
  return result;
};

node = F.DG.createNode("node");
console.log(node.getType());
node.addMember("foo", "String");
console.log(node.getMembers());
node.setCount(2);
console.log(node.getCount());
node.setData("foo", 0, "fred");
node.setData("foo", 1, "joe");
console.log(node.getData("foo", 0));
console.log(node.getData("foo", 1));

node2 = F.DG.createNode("node2");
node.setDependency( node2, "parent" );
console.log((mapNamedObjectsToNames(node.getDependencies())));
