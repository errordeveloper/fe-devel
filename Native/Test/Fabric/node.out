Node
{ foo: { name: 'foo', type: 'String' } }
2
fred
joe
{ parent: 'node2' }
